src/pages/programs/index.vue

```diff
@@ -2,71 +2,89 @@
   <div v-if="loading">
     <Loading />
   </div>
-  <div class="flex flex-col w-full h-full">
-    <div class="flex w-full p-6 mb-4 text-lg bg-white border-b border-solid shadow-sm">
-      <heroicons-outline-search class="mr-4 text-2xl text-gray-500" />
-      <input
-        v-model="searchQuery"
-        class="w-full text-lg outline-none "
-        placeholder="Rechercher"
-      >
-    </div>
 
-    <div class="flex flex-col flex-grow p-6 space-y-3">
-      <div class="flex items-center justify-between border-b">
-        <div class="flex items-center">
-          <h1 class="mb-2 text-3xl">
-            Programmes
-          </h1>
-          <Badge class="mb-1 ml-4 text-sm bg-gray-500">
-            {{ filteredPrograms.length }}
-          </Badge>
-        </div>
-        <div class="p-2 mb-2 text-white bg-green-500 rounded-full shadow cursor-pointer hover:bg-green-400">
-          <heroicons-outline-plus class="text-xl" />
-        </div>
-      </div>
+  <AppLayout>
+    <template #title>
+      Programs
+    </template>
+
+    <template #actions>
+      <AppButton :color="allProgramsAreStopped() ? 'green' : ''" @click="startPrograms()">
+        <heroicons-outline-arrow-circle-up class="mr-1" />
+        {{ allProgramsAreStopped() ? "Start" : "Restart" }} all
+      </AppButton>
+      <AppButton color="red" disabled="allProgramsAreStopped" @click="stopPrograms()">
+        <heroicons-outline-x-circle class="mr-1" />
+        Stop all
+      </AppButton>
+    </template>
+
+    <AppAlert v-if="alert.show" :type="alert.type" :close-callback="closeAlertCallback">
+      {{ alert.message }}
+    </AppAlert>
 
-      <div v-if="filteredPrograms.length > 0" class="relative">
-        <transition-group
-          tag="div"
-          enter-active-class="transition duration-200 ease-out"
-          enter-from-class="transform -translate-y-6 opacity-25"
-          enter-to-class="transform translate-y-0 opacity-100"
-          leave-active-class="absolute transition duration-200 ease-in"
-          leave-from-class="transform translate-y-0 opacity-100"
-          leave-to-class="transform translate-y-6 opacity-0"
-          move-class="transition duration-300 ease"
+    <div class="flex">
+      <div class="flex w-full mb-5 border-b border-solid">
+        <heroicons-outline-search class="mt-1 ml-1 mr-4 text-xl text-gray-500" />
+        <input
+          v-model="searchQuery"
+          class="w-full mb-2 text-lg bg-white bg-opacity-0 outline-none"
+          placeholder="Rechercher"
         >
-          <Program v-for="program in filteredPrograms" :key="program.programId" v-bind="program" class="w-full" />
-        </transition-group>
       </div>
+      <component :is="gridMode ? viewGridIcon : menuIcon" class="w-10 h-10 p-2 mb-5 ml-3 text-4xl text-gray-500 bg-white rounded-md shadow cursor-pointer hover:text-gray-600 hover:bg-opacity-50" @click="toggleLayout()" />
```

Comment:
You should directly use ViewGridIcon and MenuIcon components.
Using `component` component should be preferred only when we don't know in the template what our components can be. Here it's either one icon or another, you should be using them directly.

And it's good to use a `button` when in fact you want a button. Better for accessibility and better for every other user in fact.